### ISSUE TITLE
errorText not working

### ISSUE THREAD

**Issue by oom-** [felixlucien/flutter-autocomplete-textfield#36]
Hi,
I would want to change the message of the "errorText" value of the decoration after the "textChanged" event.

Then i have something like this: 
```js
return AutoCompleteTextField<String>(textChanged: (text) {
setState(() { if (test.length == 0) _notGood = true; else _notGood = false;},};
[...],
decoration: InputDecoration(errorText: _notGood ? 'Empty' : null,),);
```
The problem is that it's not working because the `decoration` that i pass in parameter is passed in the constructor of the `AutoCompleteTextFieldState<T>` class and like that its stay in "final" i can't do nothing to change it and the build method is not refreshing the TextField.

Here : https://github.com/felixlucien/flutter-autocomplete-textfield/blob/ff8d9ead701c2c5995c5d7e6417da33eb4adad28/lib/autocomplete_textfield.dart#L152

Is there any way to bypass this or for you to make a fix ?

Thank you !

**Comment by felixlucien** [comment]
Could you perhaps change the theme associated with the TextField in order to change colors associated with it? (By wrapping the AutoCompleteTextField in a Theme widget).

**Comment by Karlheinzniebuhr** [comment]
Having the same issue. AutoCompleteTextField does not rebuild even though I update the errorText value in setState()

**Comment by felixmccuaig** [comment]
I'll look into it.

**Comment by felixmccuaig** [comment]
The solution that I see to this is to rebuild the whole TextField with new decoration.

**Comment by felixmccuaig** [comment]
This should be fixed in 1.7.3. When changing decoration etc, use updateDecoration()

**Comment by Karlheinzniebuhr** [comment]
> This should be fixed in 1.7.3. When changing decoration etc, use updateDecoration()

I'm sorry I don't understand what you mean, I tried rebuilding the TextField and the whole AutoCompleteTextField and it didn't work for me

**Comment by XF9** [comment]
I just encountered the same issue. I solved it by wrapping `SimpleAutoCompleteTextField` and using the mentioned `updateDecoration()` if there is an current state aka not on the first draw. I'm not sure if thats  the intended way, but it works:

```dart
class SimpleAutoCompleteTextFormField extends FormField<String> {
  SimpleAutoCompleteTextFormField({
    FormFieldValidator<String> validator,
    GlobalKey<AutoCompleteTextFieldState<String>> automcompleteKey,
    InputDecoration decoration
    [..]
  }) : super(
    validator: validator,
    [..]
    builder: (FormFieldState<String> state) {
      final autoCompleteField = SimpleAutoCompleteTextField(
        key: automcompleteKey,
        decoration: decoration
        [..]
      );
      if(automcompleteKey.currentState != null)
        autoCompleteField.updateDecoration(
          decoration: decoration
        );
    }
    return autoCompleteField;
  }
);
```

Full, working, example: https://gitlab.com/snippets/1922929
